public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -670,16 +669,25 @@ function getAutoAssignedDSRef(
     return elementMapReferences.panels.get(elementKey) || new Set();
   }
 
-  return elementMapReferences.variables;
+  if (type === 'variables') {
+    return elementMapReferences.variables;
+  }
+
+  if (type === 'annotations') {
+    return elementMapReferences.annotations;
+  }
+
+  // if type is not panels, annotations, or variables, throw error
+  throw new Error(`Invalid type ${type} for getAutoAssignedDSRef`);
```

Comment:
Would you add a test case for this error? Just to ensure the error is thrown and this is not removed in the future :)